src/effects/deeplearning/ModelCard.cpp

```diff
@@ -0,0 +1,340 @@
+void ModelCard::Serialize(Writer &writer) const
+{
+   // name
+   writer.String("name");
+   writer.String(m_name.c_str());
+
+   // author
+   writer.String("author");
+   writer.String(m_author.c_str());
+   
+   // long description
+   writer.String("long_description");
+   writer.String(m_long_description.c_str());
+
+   // short description
+   writer.String("short_description");
+   writer.String(m_short_description.c_str());
+
+   // sample rate
+   writer.String("sample_rate");
+   writer.Int(m_sample_rate);
+
+   // multichannel
+   writer.String("multichannel");
+   writer.Bool(m_multichannel);
+
+   // effect type
+   writer.String("effect_type");
+   writer.String(m_effect_type.c_str());
+
+   // domain tags
+   writer.String("domain_tags");
+   writer.StartArray();
+   for (auto tag : m_domain_tags)
+      writer.String(tag.c_str());
+   writer.EndArray();
+
+   // tags
+   writer.String("tags");
+   writer.StartArray();
+   for (auto tag : m_tags)
+      writer.String(tag.c_str());
+   writer.EndArray();
+
+   // labels
+   writer.String("labels");
+   writer.StartArray();
+   for(auto label : m_labels)
+      writer.String(label.c_str());
+   writer.EndArray();
+
+   // model size
+   writer.String("model_size");
+   writer.Uint64((uint64_t)m_model_size);
+
+}
+
+void ModelCard::Deserialize(DocHolder doc, DocHolder schema)
+{
+   using namespace validators;
+   Validate(doc, schema);
+
+   // these fields are not in HF metadata but rather added later,
+   // so don't throw if they are not present
+   try
+   {
+      m_author = tryGetString("author", doc);
+      m_name = tryGetString("name", doc);
+      m_model_size = (size_t)tryGetUint64("model_size", doc);
+   }
+   catch (const InvalidModelCardDocument &e)
+   {
+      wxLogDebug(e.what());
+   }
+
+   m_long_description = tryGetString("long_description", doc);
+   m_short_description = tryGetString("short_description", doc);
+   m_effect_type = tryGetString("effect_type", doc);
+   m_domain_tags = tryGetStringArray("domain_tags", doc);
+   m_tags = tryGetStringArray("tags", doc);
+   m_labels = tryGetStringArray("labels", doc);
+   m_sample_rate = tryGetInt("sample_rate", doc);
+   m_multichannel = tryGetBool("multichannel", doc);
+}
+
+std::string ModelCard::GetRepoID() const
+{
+   return this->author() + '/' + this->name();
+}
+
+bool ModelCard::operator==(const ModelCard& that) const
+{
+    return (*this).GetRepoID() == that.GetRepoID();
+} 
+
+bool ModelCard::operator!=(const ModelCard& that) const
+{
+    return !((*this) == that);
+}
+
+// ModelCardCollection implementation
+
+ModelCardCollection::ModelCardCollection()
+{
+}
+
+void ModelCardCollection::Insert(ModelCardHolder card)
+{
+   // a lambda that dereferences the ModelCardHolder
+   // and checks if the contents are equal
+   auto isSame = [&](ModelCardHolder a)
+   {
+      return (*card) == (*a);
+   };
+   
+    // only add it if its new
+   auto it = std::find_if(this->begin(), this->end(), isSame);
```

Comment:
You can write the lambda inline!

```
auto it = std::find_if(this->begin(), this->end(), [&](ModelCardHolder a) {
      return (*card) == (*a);
});
```

Reply:
https://github.com/hugofloresgarcia/audacity/commit/c28a4cce86556566fd7c4d10829014828b225ad9